src/controllers/did.ts

```diff
@@ -543,18 +565,21 @@ export class DIDController {
 	 *         $ref: '#/components/schemas/InternalError'
 	 */
 	public async getDids(request: Request, response: Response) {
+		// Extract did from params
+		const { did } = request.params as GetDIDRequestParams;
 		// Get strategy e.g. postgres or local
 		const identityServiceStrategySetup = new IdentityServiceStrategySetup(response.locals.customer.customerId);
+
 		try {
-			const did = request.params.did
+			const didDocument  = did
 				? await identityServiceStrategySetup.agent.resolveDid(request.params.did)
 				: await identityServiceStrategySetup.agent.listDids(response.locals.customer);
 
-			return response.status(StatusCodes.OK).json(did);
+			return response.status(StatusCodes.OK).json(didDocument satisfies ListDidsResponseBody | QueryDidResponseBody);
```

Comment:
There's a relationship between `ListDidsResponseBody`, `QueryDidResponseBody`, therefore I'd make those derived from each other, 1 is declared the other one references the other, with the latter `QueryDidResponseBody` being the declaration to reference from.